src/main/java/amazin/service/AmazonService.java

```diff
@@ -0,0 +1,83 @@
+package amazin.service;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.CannedAccessControlList;
+import com.amazonaws.services.s3.model.DeleteObjectRequest;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.annotation.PostConstruct;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Date;
+
+@Service
+public class AmazonService {
+
+    private AmazonS3 s3client;
+
+    @Value("${amazonProperties.endpointUrl}")
+    private String endpointUrl;
+    @Value("${amazonProperties.bucketName}")
+    private String bucketName;
+    @Value("${amazonProperties.accessKey}")
+    private String accessKey;
+    @Value("${amazonProperties.secretKey}")
+    private String secretKey;
+
+    @PostConstruct
+    private void initializeAmazon() {
+        AWSCredentials credentials = new BasicAWSCredentials(this.accessKey, this.secretKey);
+        s3client = AmazonS3ClientBuilder
+                    .standard()
+                    .withRegion("us-east-1")
+                    .withCredentials(new AWSStaticCredentialsProvider(credentials)).build();
+    }
+
+    private String generateFileName(File file) {
+        return "pictures/" + new Date().getTime() + "-" + file.getName().replace(" ", "_");
```

Comment:
I see generalizations in this file making it seem like any file can be uploaded but we are always putting them in the pictures/ folder?

Reply:
We should only be storing pictures so storing it all in the pictures folder is fine. Please make sure the methods and variables make it clear that this is only for storing images